### ISSUE TITLE
False positive for UP037

### ISSUE THREAD

**Issue by xu-cheng** [astral-sh/ruff#17931]
### Summary

MWE to reproduce the problem:

* `__init__.py`
```py
```

* `prelude.py`
```py
__all__ = [
    "Literal",
]

from typing import Literal
```

* `test.py`
```py
from __future__ import annotations
from .prelude import *

class Foo:
    def func(
        self,
        arg: Literal["bar"],
    ) -> None:
        print(arg)
```

Ruff output:
```
$ ruff check --select=UP .
test.py:7:22: UP037 [*] Remove quotes from type annotation
  |
5 |     def func(
6 |         self,
7 |         arg: Literal["bar"],
  |                      ^^^^^ UP037
8 |     ) -> None:
9 |         print(arg)
  |
  = help: Remove quotes

Found 1 error.
[*] 1 fixable with the `--fix` option.
```

### Version

ruff 0.11.8

**Comment by Avasam** [comment]
This would require multi-file analysis: https://github.com/astral-sh/ruff/issues/7447

**Comment by ntBre** [comment]
This sounds very closely related to https://github.com/astral-sh/ruff/issues/6014, just with a different rule.
